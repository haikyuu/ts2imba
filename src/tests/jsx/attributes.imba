import {build} from "../../index"
import * as imbac from 'imba/compiler'

let tsx-code = '''
function App(){
	return (
		<input onChange={()=> {
			console.log('changed');
			let a = new Date()
			return a
		}} />
	)
}
'''

test 'jsx attributes 1' do
	const result = await build tsx-code
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()


let tsx-code2 = '''
function App(){
	return (
		<input onChange={()=> {
			console.log('changed');
		}} />
	)
}
'''

test 'jsx attributes 2' do
	const result = await build tsx-code2
	expect(result.code).toMatchSnapshot()
	const out = imbac.compile(result.code, sourceId: 'sth')
	expect(out.js).toBeDefined()
