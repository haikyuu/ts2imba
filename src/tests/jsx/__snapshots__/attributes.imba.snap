// Vitest Snapshot v1

exports[`jsx attributes 1 1`] = `
"tag App
	<self>
		<input onChange=(do
				console.log \\"changed\\"
				let a = new Date
				return a
		)>
"
`;

exports[`jsx attributes 2 1`] = `
"tag App
	<self>
		<input onChange=(do console.log \\"changed\\")>
"
`;
